sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/DoFnReflectorTest.java

```diff
@@ -533,107 +506,6 @@ public void testBadReturnType() throws Exception {
     DoFnReflector.verifyProcessMethodArguments(getClass().getDeclaredMethod("badReturnType"));
   }
```

Comment:
I worry about the loss of tests. What would you think about introducing package private parameter options for InputProvider and OutputReceiver, and using them in the tests? That allows us to keep the tests (since much of the complexity in DoFnReflector was around propagating these types, and we don't want to lose it in future refactoring)?